internal/controllers/app_controller.go

```diff
@@ -372,7 +373,7 @@ func (r *AppReconciler) reconcile(ctx context.Context, app *ketchv1.App, logger
 		}
 
 		// retry until all pods for canary deployment comes to running state.
-		if err := checkPodStatus(r.Group, r.Client, app.Name, app.Spec.Deployments[1].Version); err != nil {
+		if _, err := checkPodStatus(r.Group, r.Client, app.Name, app.Spec.Deployments[1].Version); err != nil {
```

Comment:
I see the `len(app.Spec.Deployments) <= 1` check right before this call, is there ever a case where Deployments is 0 and `app.Spec.Deployments[1].version` would result in a null pointer or as its a canary deployment we are okay?

Reply:
I added a len check. I don't see where we assure that there are deployments & assume it can't hurt to check.